New module - na_ontap_ldap_client

zeten30 opened the pull request:
##### SUMMARY
Adding new storage/netapp module -  na_ontap_ldap_client - to manage[ ONTAP ldap client configuration](https://docs.netapp.com/ontap-9/index.jsp?topic=%2Fcom.netapp.doc.exp-nfsv3-cg%2FGUID-2111ACFD-42FB-4AE7-B5CA-896751511B40.html)

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
na_ontap_ldap_client

zeten30 committed:
New module - na_ontap_ldap_client

zeten30 committed:
Fixing module example usage

zeten30 committed:
Fixing module documentation - ansible-test sanity

lonico commented on lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py:
```diff
@@ -0,0 +1,365 @@
+    required: true
+    type: str
+
+  client_config:
```
I would prefer to say name.  But I’m ok if you want to stick with this here.  Or maybe client_config_name.

lonico commented on lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py:
```diff
@@ -0,0 +1,365 @@
+        ]
+
+        # Handle required parameters for state=present
+        if self.module.params.get('state') == 'present':
```
Ansible already has required_if.   Isn’t it possible to use it here?

lonico commented on lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py:
```diff
@@ -0,0 +1,365 @@
+
+        query_details = netapp_utils.zapi.NaElement.create_node_with_children('ldap-client', **{'ldap-client-config': client_config_name})
+
+        query = netapp_utils.zapi.NaElement('query')
```
It could be safer to add vserver to the query

zeten30 replied:
I've just pushed updated code. Please check it.

lonico reviewed (approved):
shipit

zeten30 committed:
Fulfile merge request comments

- required_if used for options
- 'client_config' option renamed to 'name'
- docstring updated
- imports placed according to Ansible module best practices

lonico reviewed (approved):
shipit

zeten30 committed:
create an ems log event for users with auto support turned on

carchi8py reviewed (approved):
shipit

zeten30 commented:
Close & reopen to get rid of 'stale_ci' as suggested by ansibot :)

zeten30 committed:
Fix required_if state=present options

thedoubl3j reviewed (approved):
module is coming under netapp certified umbrella. approved by maintainers @lonico and @carchi8py. shipit
